src/main/java/com/example/java/test/junior/developer/configuration/SecurityConfig.java

```diff
@@ -0,0 +1,32 @@
+package com.example.java.test.junior.developer.configuration;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+import org.springframework.web.client.RestTemplate;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+  private final RestTemplate restTemplate;
+  @Value("${external.api.url}")
+  private String externalApiUrl;
+
+  public SecurityConfig(RestTemplate restTemplate) {
+    this.restTemplate = restTemplate;
+  }
+
+  @Override
+  protected void configure(HttpSecurity http) throws Exception {
+    http.csrf().disable()
+        .authorizeRequests()
+        .antMatchers("/api/**").authenticated()
+        .and()
+        .addFilterBefore(new JwtAuthorizationFilter(restTemplate, externalApiUrl),
```

Comment:
As I mentioned above - you don't need any filters. 
Add dependency on `org.springframework.boot:spring-boot-starter-oauth2-resource-server`
In `application.properties` set:
`spring.security.oauth2.jwt.jwk-set-uri=${KEYCLOAK_URL}/realms/app/protocol/openid-connect/certs`
Then instead of all this `.and().addFilterBefore` you just need to mention `.oauth2ResourceServer(OAuth2ResourceServerConfigurer::jwt)`
And that's it. Spring will do everything for you, no need in custom filters.